### DISCUSSION TITLE
Plug n Play vs Self hosted

### DISCUSSION THREAD

**Discussion by JakubMuzykPragmatic** [Web3Auth/Web3Auth#834] [Help]
Hi!

I'm reading the docs for some time now and I'm still confused what is the difference between paid Plug and Play web3auth version and free Self Hosted one. Please correct me if I'm wrong:
- paid version has UI for loggin and registration and self hosted does not
- for self hosted version I need to get clientId from every provider and in paid version I only need to create it in web3auth dashboard?
- what about blockchain integrations? In self hosted version I need to use libraries like web3.js or ethers.js and in paid I can connect to blockchain directly?
- are wallet adapters available only in paid version? What are they btw? Just wrappers of original wallet clients to put them in the web3auth UI? 
- are there any other differences?

Thanks in advance

**Comment by tanguyenvn** [comment]
- I think the differences between the 2 versions is that with Web3auth Plug & Play: you can easily integrate to your app within minutes but your app needs to interact with app.openlogin.com either via popup or redirect. With Self-hosted version: you have full control over UI, UX but it can take 2 weeks min of development.
- Both versions are free if your app only has < 1000 MAU and you only start paying after that. You need to create a project in Developer Dashboard and use the provided clientID in SDK. For more info of pricing, look at https://web3auth.io/pricing
- Provider ClientID in Self-hosted version: Yes, you need to register different clientIDs for login providers you want to use in your app as we need to config your app domain as redirectURI in these login providers.
- Blockchain integrations: Once the user is authenticated, the SDK returns a provider. Using this provider we can sign transactions and make RPC calls to any blockchain (ref: https://web3auth.io/docs/connect-blockchain/)
- Wallet adapters: Adapters are essentially connectors between Web3Auth and the underlying wallet provider. For example, an adapter for connecting with torus wallet is available under web3auth as [@web3auth/torus-evm-adapter](https://web3auth.io/docs/sdk/web/adapters/torus-evm). Every adapter follows a common interface which is required by Web3Auth to communicate with the wallet Ref https://web3auth.io/docs/sdk/web/adapters/
